codeflash/optimization/function_optimizer.py

```diff
@@ -405,6 +429,33 @@ def determine_best_candidate(
                         future_line_profile_results = None
                     candidate_index += 1
                     candidate = candidates.popleft()
+
+                    formatted_candidate_code = candidate.source_code
+                    if self.args.formatter_cmds:
+                        temp_code_file_path: Path | None = None
+                        try:
+                            with tempfile.NamedTemporaryFile(
+                                mode="w",
+                                suffix=".py",
+                                delete=False,
+                                encoding="utf8",
+                                dir=self.optimizer_temp_dir
+                            ) as tmp_file:
+                                tmp_file.write(candidate.source_code)
+                                temp_code_file_path = Path(tmp_file.name)
+                            
+                            formatted_candidate_code = format_code(
```

Comment:
formatting code is depenedent on the cwd of the code, imports are grouped according to what module they belong to, for the project's own module they are grouped together. This determination of what is the module they belong to is determined by the cwd.
So we should not format code in a temp directory, the results may not be the same. This is btw why your unit tests were failing today

Reply:
I don't think that's true, as this is working.

This is only formatting the function snippet and the helper snippets -- I don't think this has to do with why those unit tests are failing.